lib/hydra/works/services/characterization_service.rb

```diff
@@ -63,9 +57,10 @@ def extract_metadata(content)
         end
       end
 
+      # Generate a random file name
       def temp_file_name
-        m = %r{/([^/]*)$} .match(object.uri)
-        "#{m[1]}-content.tmp"
+        m = (0...8).map { (65 + rand(26)).chr }.join
```

Comment:
Originally, it was filename created using the uri. However, uri doesn't exist yet because the content hasn't been saved. Maybe it should just already exist? I can change the service around to raise an error if it doesn't... You can still pass in a file in order to avoid re-downloading the content from Fedora.